### PR TITLE
Remove shared notifications infrastructure

### PR DESCRIPTION
**Pull request by timbussmann**
based on https://github.com/Particular/NServiceBus/pull/5454

introduce the `Notifications` class which allows every component to manage events themselves

### CODE REVIEW

File: src/NServiceBus.Core/Features/FeatureConfigurationContext.cs [hunk 1]
```diff
@@ -41,6 +42,8 @@ internal FeatureConfigurationContext(ReadOnlySettings settings, IConfigureCompon
 
         internal RoutingComponent Routing { get; }
 
+        internal RecoverabilityComponent Recoverability { get; }
```
Comment (SeanFeldman) [inline src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: Is this used anywhere?

Comment (timbussmann) [reply src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: yes here for the acceptance testing framework: https://github.com/Particular/NServiceBus/pull/5503/files#diff-e6d8bf51552c83a965bcd54d6f1ea308L26

File: src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs [hunk 2]
```diff
@@ -18,7 +18,9 @@ class RecoverabilityComponent
         public RecoverabilityComponent(SettingsHolder settings)
         {
             this.settings = settings;
-
+            var configuration = settings.Get<Configuration>();
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs]: The other components takes this as an argument to not use the settings inside the component but since this component is not yet aligned we should do that in a separate PR

**Comment by timbussmann** [comment]
rebased

File: src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs (hunk 2 above)
Comment (timbussmann) [reply src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs]: yeah that's the reason I did it this way for now

File: src/NServiceBus.Core/Features/FeatureConfigurationContext.cs (hunk 1 above)
Comment (timbussmann) [reply src/NServiceBus.Core/Features/FeatureConfigurationContext.cs]: I actually have some ideas how we could avoid this. I'll raise a separate PR though.
